File: regions/system/library/csharp/schemaregistry/SchemaRegistryConfig.cs
namespace K1s0.System.SchemaRegistry;

public sealed record SchemaRegistryConfig
{
    public required string Url { get; init; }

    public string? Username { get; init; }

    public string? Password { get; init; }

    public CompatibilityMode CompatibilityMode { get; init; } = CompatibilityMode.Backward;

    public static string SubjectName(string topic, string suffix = "value") => $"{topic}-{suffix}";
}

File: regions/system/library/swift/schemaregistry/Sources/K1s0SchemaRegistry/SchemaRegistryConfig.swift
import Foundation

/// Schema Registry 接続設定。
public struct SchemaRegistryConfig: Sendable {
    public let url: String
    public let compatibility: CompatibilityMode
    public let timeoutSeconds: TimeInterval

    public init(
        url: String,
        compatibility: CompatibilityMode = .backward,
        timeoutSeconds: TimeInterval = 30
    ) {
        self.url = url
        self.compatibility = compatibility
        self.timeoutSeconds = timeoutSeconds
    }

    /// Confluent 規則に従ったサブジェクト名を返す。
    ///
    /// 規則: `{topic}-value`
    public static func subjectName(for topic: String) -> String {
        "\(topic)-value"
    }
}
